infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -557,7 +566,7 @@ private String getDataCorruptionErrorMessage(
      * @return ByteBuffer for the payload
      * @throws IOException IO exception
      */
-    private ByteBuffer getPayloadForMetadata(FileChannel fileChannel, Metadata metadata) throws IOException {
+    public static ByteBuffer getPayloadForMetadata(FileChannel fileChannel, Metadata metadata) throws IOException {
```

Comment:
Hi Viacheslav, likewise. In this scenario, it's because this method is not used outside of this class right?

Reply:
hi @xnull - this method does not use any instance variables, so I was the one who suggested making it as public static to avoid copying the code and re-using this method. 
Please explain why that is not correct and suggest a simpler alternative so we can re-use functions like these? thanks

Reply:
I think @xnull is correct about the method `getPayloadForMetadata`, since this is not being used outside of this class.